### PR TITLE
Skip removing Psalm PhpDocTagNode

### PR DESCRIPTION
**Pull request by rajyan**
closes https://github.com/rectorphp/rector-src/pull/1799

### CODE REVIEW

**Comment by rajyan** [comment]
@leighman 

Hi, I cherry-picked your test case and tried to fix it.

**Comment by rajyan** [comment]
@samsonasik 
https://github.com/rectorphp/rector-src/commit/39edeb3f2d5a890ed5472c278247329f579f3737
This is a similar situation to this commit.

I changed to totally ignore psalm prefixed annotations because it's most time not intended to remove. What do you think about this change?

File: packages/BetterPhpDocParser/PhpDocInfo/PhpDocInfo.php [hunk 1]
```diff
@@ -305,7 +305,7 @@ public function removeByType(string $typeToRemove): void
             $typeToRemove
         ): ?int {
             if ($node instanceof PhpDocTagNode && is_a($node->value, $typeToRemove, true)) {
-                if ($typeToRemove === VarTagValueNode::class && $node->name !== '@var') {
+                if (str_starts_with($node->name, '@psalm-')) {
```
Comment (staabm) [inline packages/BetterPhpDocParser/PhpDocInfo/PhpDocInfo.php]: Should this also skip "phpstan-" prefixed ones?

Comment (rajyan) [reply packages/BetterPhpDocParser/PhpDocInfo/PhpDocInfo.php]: Thanks!
I looked into https://github.com/phpstan/phpdoc-parser/blob/d5a6692b8a74aeb00cc25e144f37a35da2bc31de/src/Parser/PhpDocParser.php#L121-L206
and decided to add "phpstan-" prefix too.

**Review by samsonasik — APPROVED** [review]

**Comment by samsonasik** [comment]
Thank you @rajyan @staabm

**Pull request merged**